config/v1/types_infrastructure.go

```diff
@@ -457,6 +462,32 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+type ClusterDNSConfig struct {
+	// APIServerDNSConfig contains DNS configuration for the API Server.
```

Comment:
What happens when this is omitted? What's the expectation?

Reply:
When the userConfiguredDNS feature is enabled (via install-config), the Installer and not the user is responsible for populating these values. Even with this feature enabled, the Installer is responsible for configuring the external and internal Load Balancers for API and API-Int servers respectively and utilizes the LB DNS names to derive the DNS configuration.

So, these values will be omitted only when the feature is Disabled (which is the default). If they are empty when the userConfiguredDNS is Enabled, then cluster cannot be created and it is a failure scenario.

Does any of this need to go into the godoc?

Reply:
So, if this field is provided, then we expect all three types of DNS to be specified? Just trying to work out if it's possible to have 
```
apiServerDNSConfig: []
internalAPIServerDNSConfig:
- ....
ingressDNSConfig: []
```
It's making me thinking that actually, these fields shouldn't be optional and that each one needs an entry when the struct is specified.

Reply:
Some of this context would be useful in the parent, where the ClusterDNSConfig is included, we could have godoc there that explains this field is expected to be omitted when the user configured DNS feature wasn't used in the installation process, and mention what it means for these values to be present there